test/e2e/data/infrastructure-docker/v1beta1/clusterclass-quick-start.yaml

```diff
@@ -32,6 +32,52 @@ spec:
             apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
             kind: DockerMachineTemplate
             name: quick-start-docker-worker-machinetemplate
+  variables:
```

Comment:
[TBD] if this is a good idea. I think it's nice to have some basic ClusterClass JSON patching e2e test coverage with the ClusterClass e2e test.

What I don't like about it is that the current version duplicates CAPD default values here.

Reply:
@fabriziopandini I'm totally fine with dropping this from this PR if we think it should be added via another test / or via a separate PR. WDTY?

(P.S. I think we'll talk about testing anyway)

Reply:
I'm ok in having this in this PR, it is just two yaml files with examples

Reply:
What about using this to compute the docker images, so we are testing also builtin variables

Reply:
> I'm ok in having this in this PR, it is just two yaml files with examples

Just to avoid misunderstandings, those files are used in our quickstart e2e test.

> What about using this to compute the docker images, so we are testing also builtin variables

Done, works! 
I also tested a Kubernetes upgrade manually and it also works there, i.e. the Machines get the correct customImage version. (had to rebase on top of main to pick up your mergepatch helper bugfix)